### PR TITLE
[MM-9563] Add a red dot to browser favicon when there are unread mentions

### PR DESCRIPTION
**Pull request by jkl5616**
#### Summary
- Display red dot favicon when there are unread mentions
- Add red dot favicons of 16x16, 32x32, 96x96 sizes

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/9502

#### Screenshot
screenshot

### CODE REVIEW

File: components/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -221,7 +225,7 @@ export default class Sidebar extends React.PureComponent {
         this.updateTitle();
 
         // Don't modify favicon for now: https://mattermost.atlassian.net/browse/MM-13643.
```
Comment (hmhealey) [inline components/sidebar/sidebar.jsx]: Can you remove this comment since it's incorrect now?

File: components/sidebar/sidebar.jsx [hunk 2]
```diff
@@ -237,26 +241,18 @@ export default class Sidebar extends React.PureComponent {
     }
 
     setBadgesActiveAndFavicon() {
+        const link = document.querySelector('link[rel="icon"]');
+
+        if (!link) {
```
Comment (hmhealey) [inline components/sidebar/sidebar.jsx]: Maybe we could check this inside `this.updateFavicon` since that's where we actually modify those links?

Comment (jkl5616) [reply components/sidebar/sidebar.jsx]: Well, If I move this into `this.updateFavicon`, `this.badgesActive` will be updated even if there are no favicon links in the document, which can cause inconsistency between `this.badgesActive` and what favicons are really representing.
So I put that code before updating `this.badgesActive`.

File: components/sidebar/sidebar.jsx (hunk 1 above)
Comment (jkl5616) [reply components/sidebar/sidebar.jsx]: Sure

File: components/sidebar/sidebar.jsx (hunk 2 above)
Comment (hmhealey) [reply components/sidebar/sidebar.jsx]: That's a good point. I don't think we can actually get into a case where those links don't exist, but it doesn't hurt to keep this around

File: components/sidebar/sidebar.test.jsx [hunk 3]
```diff
@@ -507,4 +507,27 @@ describe('component/sidebar/sidebar_channel/SidebarChannel', () => {
...
+        instance.updateFavicon = jest.fn();
+
+        wrapper.setProps({unreads: {mentionCount: 0, messageCount: 0}});
+        instance.setBadgesActiveAndFavicon();
+        wrapper.setProps({unreads: {mentionCount: 3, messageCount: 4}});
+        instance.setBadgesActiveAndFavicon();
+        expect(instance.updateFavicon).lastCalledWith(true);
+
+        wrapper.setProps({unreads: {mentionCount: 0, messageCount: 4}});
+        instance.setBadgesActiveAndFavicon();
```
[the remark this answers is not in the document]
Comment (jkl5616) [reply components/sidebar/sidebar.test.jsx]: @saturninoabril Thanks for the review! Please check for the changes.

**Comment by jkl5616** [comment]
@saturninoabril Ready to merge!

**Comment by saturninoabril** [comment]
Thank you @jkl5616 for your contribution!  Will merge after passing the build.

**Pull request merged**